src/fitnesse/slim/fixtureInteraction/DefaultInteraction.java

```diff
@@ -3,88 +3,144 @@
+		Object[] initargs = ConverterSupport.convertArgs(args,
+				constructor.getParameterTypes());
+
+		return newInstance(constructor, initargs);
+	}
+
+
+	protected Class<?> searchPathsForClass(List<String> paths, String className) {
+		Class<?> k = getClass(className);
+
+		if (k != null) {
+			return k;
+		}
+		for (String path : paths) {
+			k = getClass(path + "." + className);
+			if (k != null) {
+				return k;
+			}
+		}
+		throw new SlimError(String.format("message:<<%s %s>>", SlimServer.NO_CLASS, className));
+	}
+
+	protected Class<?> getClass(String className) {
+		try {
+			return Class.forName(className);
+		} catch (ClassNotFoundException e) {
+			return null;
+		}
+	}
+
+	protected Constructor<?> getConstructor(Class<?> clazz,
+			Object[] args) {
+
+		for (Constructor<?> constructor : clazz.getConstructors()) {
+			Class<?>[] arguments = constructor.getParameterTypes();
+			if (arguments.length == args.length) {
+				return constructor;
+			}
+		}
+		return null;
+	}
+
+	protected Object newInstance(Constructor<?> constructor, Object... initargs) throws InvocationTargetException, InstantiationException, IllegalAccessException {
+		return constructor.newInstance(initargs);
+	}
+
+	@Override
+	public MethodExecutionResult findAndInvoke(String methodName, Object instance, Object... args) throws Throwable {
+		Method method = findMatchingMethod(methodName, instance.getClass(), args.length);
+		if (method != null) {
+			return this.invokeMethod(instance, method, args);
+		}
+		return MethodExecutionResult.noMethod(methodName, instance.getClass(), args.length);
+	}
+	
+	protected Method findMatchingMethod(String methodName, Class<?> k, int nArgs) {
+		Method[] methods = k.getMethods();
+		if(methods == null) {
+			methods = k.getMethods();
+		}
+
+		for (Method method : methods) {
+			boolean hasMatchingName = method.getName().equals(methodName);
+			boolean hasMatchingArguments = method.getParameterTypes().length == nArgs;
+			if (hasMatchingName && hasMatchingArguments) {
+				return method;
+			}
+		}
+		return null;
+	}
+
+	protected MethodExecutionResult invokeMethod(Object instance, Method method, Object[] args) throws Throwable {
+		Object[] convertedArgs = convertArgs(method, args);
+		Object retval = callMethod(instance, method, convertedArgs);
+		Class<?> retType = method.getReturnType();
+		return new MethodExecutionResult(retval, retType);
+	}
+
+	protected Object[] convertArgs(Method method, Object[] args) {
+		Type[] argumentParameterTypes = method.getGenericParameterTypes();
+		return ConverterSupport.convertArgs(args, argumentParameterTypes);
+	}
+
+	protected Object callMethod(Object instance, Method method, Object[] convertedArgs) throws Throwable {
+		try {
+			Object result;
+			if (instance instanceof InteractionAwareFixture) {
+				// invoke via interaction, so it can also do its thing on the aroundMethod invocation
+				result = ((InteractionAwareFixture)instance).aroundSlimInvoke(this, method, convertedArgs);
```

Comment:
In the current release the fixtureInteraction only does the invokeMethod, with 'find' living in the `MethodExecutor`. I believe that's what you wanted to address  (and I think is a good idea) :-)

I believe the current setup will work, and although less clear than I would ideally want it's not too bad (and adding an extra layer to separate but keep together the find and invoke parts creates new complexity also). And this gives people the flexibility to do what they need.

I would still prefer it the calling of 'aroundSlimInvoke()' would use `methodInvoke()`, as it did before: to ensure that any overriding of `methodInvoke` also sees/captures the call to `aroundSlimInvoke`.